test/drop_test/ffi.py

```diff
@@ -480,27 +497,33 @@ def start(self, addr: str, dbpath: str, max_reqs_per_sec: int):
         )
         if err != 0:
             err_type = LibResult(err).name
-            raise Exception(f"norddrop_start has failed with code: {err}({err_type})")
+            raise DropException(
+                f"norddrop_start has failed with code: {err}({err_type})", err
+            )
 
     def stop(self):
         err = self._lib.norddrop_stop(self._instance)
         if err != 0:
             err_type = LibResult(err).name
-            raise Exception(f"norddrop_stop has failed with code: {err}({err_type})")
+            raise DropException(
+                f"norddrop_stop has failed with code: {err}({err_type})", err
+            )
 
     @property
     def version(self) -> str:
         version = self._lib.norddrop_version(self._instance)
         if not version:
-            return ""
+            raise DropException(f"norddrop_version has failed")
 
         return ctypes.string_at(version).decode("utf-8")
 
     def __del__(self):
         err = self._lib.norddrop_destroy(self._instance)
         if err != 0:
             err_type = LibResult(err).name
-            raise Exception(f"norddrop_destory has failed with code: {err}({err_type})")
+            raise DropException(
+                f"norddrop_destory has failed with code: {err}({err_type})", err
+            )
```

Comment:
Is it OK to throw in the destructor in Python?

Reply:
Honestly, no idea didn't even think about it. Quick googling doesn't reveal anything as well. Do you have any idea about possible side effects?

Reply:
No, but I'm not that deep into Python. In Rust or C++ it's a severe offense

Reply:
I also have no idea about python, can't really find anything. I thought about maybe doing the creation and destructor explicitly but that doesn't seem worth it to me as that's only done once in the app lifecycle

Reply:
Yeah, so let's leave it as is. We will continue the discussion when it breaks in any way